autora/cycle/plot_utils.py

```diff
@@ -0,0 +1,470 @@
+    """
+    Gets theory predictions over conditions space and saves results of each cycle to a dictionary.
+    Args:
+        cycle: AER Cycle object that has been run
+        conditions: Condition space. Should be an array of grouped conditions.
+
+    Returns: dict
+
+    """
+    d_predictions = {}
+    for i, theory in enumerate(cycle.data.theories):
+        d_predictions[i] = theory.predict(conditions)
+
+    return d_predictions
+
+
+def check_replace_default_kw(default: dict, user: dict) -> dict:
+    """
+    Combines the key/value pairs of two dictionaries, a default and user dictionary. Unique pairs
+    are selected and user pairs take precedent over default pairs if matching keywords. Also works
+    with nested dictionaries.
+
+    Returns: dict
+    """
+    # Copy dict 1 to return dict
+    d_return = default.copy()
+    # Loop by keys in dict 2
+    for key in user.keys():
+        # If not in dict 1 add to the return dict
+        if key not in default.keys():
+            d_return.update({key: user[key]})
+        else:
+            # If value is a dict, recurse to check nested dict
+            if isinstance(user[key], dict):
+                d_return.update(
+                    {key: check_replace_default_kw(default[key], user[key])}
+                )
+            # If not a dict update the default value with the value from dict 2
+            else:
+                d_return.update({key: user[key]})
+
+    return d_return
+
+
+def plot_results_panel_2d(
+    cycle: Cycle,
+    iv_name: Optional[str] = None,
+    dv_name: Optional[str] = None,
+    steps: int = 50,
+    wrap: int = 4,
+    spines: bool = False,
+    subplot_kw: dict = {},
+    scatter_kw1: dict = {},
+    scatter_kw2: dict = {},
+    line_kw: dict = {},
+) -> plt.figure:
+    """
+    Generates a multi-panel plot with each panel showing results of an AER cycle. Observed data
+    is plotted as a scatter plot with the current cycle colored differently than observed data from
+    previous cycles. The current cycle's theory is plotted as a line over the range of the observed
+    data.
+    Args:
+        cycle: AER Cycle object that has been run
+        iv_name: Independent variable name. Name should match the name instantiated in the cycle
+                    object. Default will select the first.
+        dv_name: Single dependent variable name. Name should match the names instantiated in the
+                    cycle object. Default will select the first DV.
+        steps: Number of steps to define the condition space to plot the theory.
+        wrap: Number of panels to appear in a row. Example: 9 panels with wrap=3 results in a
+                3x3 grid.
+        spines: Show axis spines for 2D plots, default False.
+        subplot_kw: Dictionary of keywords to pass to matplotlib 'subplot' function
+        scatter_kw1: Dictionary of keywords to pass to matplotlib 'scatter' function that plots the
```

Comment:
maybe rename this parameter:
`existing_data_kw`

Reply:
Yeah I thought about naming it that way, though I'm a bit conflicted with removing the contextual link to the underlying Matplotlib function. Maybe something like `existing_data_scatter_kw`? Is that too verbose?